syntax = "proto3";
package model.detections.raw;


message IntCoordinate {
    int32 x = 1;
    int32 y = 2;
}

message FloatCoordinate {
    float x = 1;
    float y = 2;
}

message BBox {
    IntCoordinate top_left = 1;
    int32 width = 2;
    int32 height = 3;
}

message OffsetVector {
    FloatCoordinate position = 1;
    float magnitude = 2;
}

message HandDetection {
    enum HandState {
        NO_CONTACT = 0;
        SELF_CONTACT = 1;
        ANOTHER_PERSON = 2;
        PORTABLE_OBJECT = 3;
        STATIONARY_OBJECT = 4;
    }

    enum HandSide {
        LEFT = 0;
        RIGHT = 1;
    }

    BBox bbox = 1;
    float score = 2;
    HandState state = 3;
    OffsetVector offset = 4;
    HandSide side = 5;
}

message ObjectDetection {
    BBox bbox = 1;
    float score = 2;
}

message Detections {
    string video_id = 1;
    int32 frame_number = 2;
    repeated HandDetection hands = 3;
    repeated ObjectDetection objects = 4;
}
